Geometry/MuonCommonData/plugins/dd4hep/DDGEMAngular.cc

```diff
@@ -15,11 +15,11 @@ static long algorithm(dd4hep::Detector& /* description */,
   cms::DDAlgoArguments args(ctxt, e);
 
   // Header section of original DDGEMAngular.h
-  float startAngle = args.value<float>("startAngle");
-  float stepAngle = args.value<float>("stepAngle");
+  double startAngle = args.value<double>("startAngle");
```

Comment:
@bsunanda - whenever a rotation matrix is involved it always uses `double`:
https://cmssdt.cern.ch/lxr/source/DetectorDescription/DDCMS/interface/DDAlgoArguments.h#0022

Reply:
But we noticed that if we have these operations in floats we come up with overlaps (small but it matters)

Reply:
ok, I see its probably the result of a `cos(phitmp)` and ` rPos * sin(phitmp)` on line 61. You could test it by defining `phitmp` as `double`.

Reply:
I would prefer if all definitions of phi's are done in double. Addition in double does not cost much

Reply:
no, it can be up to 20% depending on an algorithm. There was a report at a Sim meeting a while ago.

Reply:
I think there are two options for how to change this code.
1. Change only the cos/sin computation line.
2. Make larger changes of converting `float` to `double`, but also tackle performance testing and improvements. The `for` loop  in lines 47-70 has several lines that are needlessly performed for each iteration.
@bsunanda Do you want to limit this PR to only fixing the overlaps, or would you like to include performance improvements in this PR, too?

Reply:
Hi all,

our primary goal is correctness, so 20% CPU means nothing compare to situation when results cannot be accepted. I would remove all floats without any test.

Reply:
I believe avoiding overlap is more critical. One can check the performance issue later on.

Reply:
> Hi all,
> 
> our primary goal is correctness, so 20% CPU means nothing compare to situation when results cannot be accepted. I would remove all floats without any test.

Please, note, it's also memory consumption, not just CPU.